vars/commonstages.groovy

```diff
@@ -14,24 +14,35 @@ def deleteExistingHelmChart() {
     }
 }
 // Function to get the service version
-def setServiceVersion(String upstreamEnv) {
-    withFolderProperties {
-        container('jenkins-agent') {
-            script {
-                if("${ENVIRONMENT_NAME}" == "sit") {
-                    def app = sh(returnStdout: true, script: "curl -k 'https://artifactory.test.cicd.com/artifactory/anurag-generic/services-build-versions/service/version.txt'")
-                    env.SERVICE_VERSION = app.split('=')[1]
-                }
-                else {
-                    def upstream_build_number = UPSTREAM_BUILD.split('#')[1]
-                    def app = sh(returnStdout: true, script: "curl -k https://artifactory.test.cicd.com/artifactory/anurag-generic/${upstreamEnv}-verified/service/version-${upstream_build_number}.txt")
-                    env.SERVICE_VERSION = app.split('=')[1]
-                }
-            }
+import groovyx.net.http.HTTPBuilder
+
+def setServiceVersion() {
+    // Create an HTTPBuilder instance
+    def http = new HTTPBuilder()
+
+    try {
+        // Check if the environment is "sit"
+        if("${ENVIRONMENT_NAME}" == "sit") {
+            // If it is, get the service version from a URL
+            def response = http.get("https://artifactory.test.cicd.com/artifactory/anurag-generic/services-build-versions/service/version.txt")
+            env.SERVICE_VERSION = response.split('=')[1]
```

Comment:
Can you save the response as a json , and then get service version from json using grrovy ?